Bugzilla/Hook.pm

```diff
@@ -55,6 +56,71 @@ sub in {
     return $hook_name eq $currently_in ? 1 : 0;
 }
 
+
+sub template_process {
+    my ($context, $hook_name, $template) = @_;
+    $template ||= $context->stash->{component}->{name};
+
+    # sanity check:
+    if (!$template =~ /[\w\.\/\-_\\]+/) {
+        ThrowCodeError('template_invalid', { name => $template });
+    }
+
+    my (undef, $path, $filename) = File::Spec->splitpath($template);
+    $path ||= '';
+    $filename =~ m/(.+)\.(.+)\.tmpl$/;
```

Comment:
It would be comforting if template filenames were more restrictive than regex ```.``` here.